test/unit/sdam/topology.test.js

```diff
@@ -457,30 +437,33 @@ describe('Topology (unit)', function () {
       // satisfy the initial connect, then restore the original method
       const selectServer = this.sinon
         .stub(Topology.prototype, 'selectServer')
-        .callsFake(function (selector, options, callback) {
+        .callsFake(function () {
           const server = Array.from(this.s.servers.values())[0];
           selectServer.restore();
-          callback(null, server);
+          return Promise.resolve(server);
         });
 
       this.sinon.stub(Server.prototype, 'connect').callsFake(function () {
         this.s.state = 'connected';
         this.emit('connect');
+        return Promise.resolve();
       });
 
-      topology.connect(() => {
-        topology.selectServer(ReadPreference.secondary, { serverSelectionTimeoutMS: 1000 }, err => {
-          expect(err).to.exist;
-          expect(err).to.match(/Server selection timed out/);
-          expect(err).to.have.property('reason');
+      topology.connect().then(() => {
+        topology
+          .selectServer(ReadPreference.secondary, { serverSelectionTimeoutMS: 1000 })
+          .then(expect.fail, err => {
+            expect(err).to.exist;
+            expect(err).to.match(/Server selection timed out/);
+            expect(err).to.have.property('reason');
 
-          // When server is created `connect` is called on the monitor. When server selection
-          // occurs `requestCheck` will be called for an immediate check.
-          expect(requestCheck).property('callCount').to.equal(1);
+            // When server is created `connect` is called on the monitor. When server selection
+            // occurs `requestCheck` will be called for an immediate check.
+            expect(requestCheck).property('callCount').to.equal(1);
 
-          topology.close();
-          done();
-        });
+            topology.close();
+            done();
```

Comment:
Still using done here we should switch to returing the promise. This test is crashing the unit runs because the call count is 5 🤔